internal/spdxlicense/generate/license.go

```diff
@@ -43,7 +60,7 @@ func (l License) canReplace(other License) bool {
 		}
 	}
 
-	return l.ID != other.ID
+	return l.ID == other.ID
```

Comment:
Why did we have to flip this here? I admit it's a little hard to reason about this case since it's the last one checked with so many other chances of an early return

Reply:
Thanks for the look @spiffcs -- this function is `canReplace`, I think we haven't seen this bug because the other negative test conditions failed first. Basically: this function is trying to find any license that could be replaced by another license by checking equality in almost all the other fields. However, when it gets to the end here, the condition previously here is "if the ID is _NOT_ the same, we can consider it equal". Two comments about this: 1) this is not ever triggered based on the current document, and 2) I really don't think this is right -- if at the end of all the checks, the IDs are the same, it would return `false`... which means _maybe_ we have different `Text`...